addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1266,6 +1270,88 @@ options.registry.parallax = options.Class.extend({
     onMove: function () {
         this._refreshPublicWidgets();
     },
+    /**
+     * @override
+     */
+    updateUI: async function () {
+        await this._super(...arguments);
+        // Remove parallax if no background-image.
+        // Example: click on the remove background-image button.
+        if (this.isParallax) {
+            let $el = this.$target.find('> .s_parallax_bg');
+            const hasBackgroundImage = $el.css('background-image') !== 'none';
+            if (!hasBackgroundImage) {
+                this._removeParallax();
+                this.$target.removeClass('oe_img_bg');
+            }
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Build/remove parallax.
+     *
+     * @see this.selectClass for parameters
+     */
+    parallax: async function (previewMode, widgetValue, params) {
+        const isParallax = widgetValue === 'true';
+        this.$target.toggleClass('parallax', isParallax);
+        await this._refreshPublicWidgets();
+        if (isParallax) {
+            this._initParallax();
+        } else {
+            this._removeParallax();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeVisibility: function () {
+        return !this.$target.hasClass('o_background_video');
+    },
```

Comment:
Actually, seeing this, I think there is something strange: you don't allow to add background video when the snippet uses parallax but you hide parallax when there is a background video. I think it should be consistent: if you add a color *or a video*, the background is removed, so the parallax effect is removed (and its UI hidden).

Reply:
ok to allow video when it's in parallax and remove the parallax but I have to keep this code to hide the option when there is a background video.

Reply:
Indeed